robottelo/hosts.py

```diff
@@ -1752,14 +1752,26 @@ def __init__(self, hostname=None, **kwargs):
         self._api = type('api', (), {'_configured': False})
         self._cli = type('cli', (), {'_configured': False})
 
+    def _swap_nailgun(self, new_version):
+        """Install a different version of nailgun from GitHub and invalidate the module cache."""
+        import sys
+
+        from pip._internal import main as pip_main
+
+        pip_main(['uninstall', '-y', 'nailgun'])
+        pip_main(['install', f'https://github.com/SatelliteQE/nailgun/archive/{new_version}.zip'])
+        self._api = type('api', (), {'_configured': False})
+        to_clear = [k for k in sys.modules.keys() if 'nailgun' in k]
+        [sys.modules.pop(k) for k in to_clear]
```

Comment:
Does this clearance should be done after uninstalling the previous Nailgun version and before installing new nailgun version ?

Reply:
This implementation would likely cleanup the bits of newly installed nailgun as well hence above comment, correct me if I am wrong .

Reply:
the order here is less important than you might think. the current modules are live in memory, so this only really helps for new imports (like what we do when recreating the `Satellite.api` attributes.

Reply:
Interesting how popping out the nailgun specific modules from sys.modules would still make nailgun module available.

Do you mean the newly installed nailgun isnt the part of `sys.modules` and only unistalled is ?